src/uwtools/rocoto.py

```diff
@@ -106,25 +106,24 @@ def realize_rocoto_xml(
     :return: Did the input and output files conform to theirr schemas?
     """
 
-    # Validate the YAML.
-    if uwtools.config.validator.validate_yaml(
-        config_file=config_file, schema_file=_rocoto_schema_yaml()
-    ):
-        # Render the template to a temporary file.
-        with tempfile.NamedTemporaryFile(delete=False) as temp_file:
-            _write_rocoto_xml(
-                config_file=config_file,
-                rendered_output=temp_file.name,
-            )
-            # Validate the XML.
-            if validate_rocoto_xml(input_xml=temp_file.name):
-                # If no issues were detected, save temp file and report success.
-                shutil.move(temp_file.name, rendered_output)
-                return True
+    if not validate_yaml(config_file=config_file, schema_file=_rocoto_schema_yaml()):
+        log.error("YAML validation errors identified in %s", config_file)
+        return False
+
+    temp_file = tempfile.NamedTemporaryFile(delete=False)  # pylint: disable=consider-using-with
```

Comment:
I contributed some of these lines so can comment.

The context manager automatically deletes the temporary file when its block ends. We could potentially move the statements below inside the block (I assume that the auto-delete won't complain if the file has already disappeared having been moved/renamed, but we'd have to check). But maybe we should use `mkstemp()` instead, which doesn't provide a context manager, so that we wouldn't need this ugly pylint suppressor. I'll work with @WeirAE to see.

Reply:
I don't think the context manager will work: It _will_ eventually exit and remove the temp file, and we want the temp file to remain on disk if it fails XML validation so that the user can inspect it.

Reply:
`mkstemp()` version has been added in the last push, with fixed tests - I like this best of the 3